src/app/containers/ConsentBanner/index.canonical.test.jsx

```diff
@@ -0,0 +1,115 @@
+const privacyOnReject = jest.fn();
+const cookieOnAllow = jest.fn();
+const cookieOnReject = jest.fn();
+
+jest.mock('./Banner', () => jest.fn());
+const Banner = require('./Banner');
+
+Banner.mockImplementation(({ type }) => <div>Canonical {type} banner</div>);
+
+describe('Canonical Container', () => {
+  beforeEach(() => {
+    jest.mock('./CanonicalLogic', () => jest.fn());
+    logic = require('./CanonicalLogic');
+
+    logic.mockImplementation(
+      ({ setShowPrivacyBanner, setShowCookieBanner }) => {
+        setShowCookieBannerState = setShowCookieBanner;
+        setShowPrivacyBannerState = setShowPrivacyBanner;
+
+        return {
+          runInitial,
+          privacyOnAllow,
+          privacyOnReject,
+          cookieOnAllow,
+          cookieOnReject,
+        };
+      },
+    );
+
+    CanonicalContainer = require('./index').default;
+
+    container = document.createElement('div');
+    document.body.appendChild(container);
+  });
+
+  it('should render privacy banner when both banners are set to be shown', () => {
+    act(() => {
+      ReactDOM.render(<CanonicalContainer />, container);
+
+      setShowPrivacyBannerState(true);
+      setShowCookieBannerState(true);
+    });
+
+    expect(container.innerHTML).toBe('<div>Canonical privacy banner</div>');
+    expect(runInitial).toHaveBeenCalled();
+    expect(logic).toHaveBeenCalledWith({
+      setShowCookieBanner: expect.any(Function),
+      setShowPrivacyBanner: expect.any(Function),
+    });
+    expect(Banner).toHaveBeenCalledWith(
+      {
+        onAccept: privacyOnAllow,
+        onReject: privacyOnReject,
+        type: 'privacy',
+      },
+      {},
+    );
+  });
+
+  it('should render privacy banner when only privacy banner is set to be shown', () => {
+    act(() => {
+      ReactDOM.render(<CanonicalContainer />, container);
+
+      setShowPrivacyBannerState(true);
+    });
+
+    expect(container.innerHTML).toBe('<div>Canonical privacy banner</div>');
+    expect(runInitial).toHaveBeenCalled();
+    expect(logic).toHaveBeenCalledWith({
+      setShowCookieBanner: expect.any(Function),
+      setShowPrivacyBanner: expect.any(Function),
+    });
+    expect(Banner).toHaveBeenCalledWith(
+      {
+        onAccept: privacyOnAllow,
+        onReject: privacyOnReject,
+        type: 'privacy',
+      },
+      {},
+    );
+  });
+
+  it('should render cookie banner when only cookie banner is set to be shown', () => {
+    act(() => {
+      ReactDOM.render(<CanonicalContainer />, container);
+
+      setShowCookieBannerState(true);
+    });
+
+    expect(container.innerHTML).toBe('<div>Canonical cookie banner</div>');
+    expect(runInitial).toHaveBeenCalled();
+    expect(logic).toHaveBeenCalledWith({
+      setShowCookieBanner: expect.any(Function),
+      setShowPrivacyBanner: expect.any(Function),
+    });
+    expect(Banner).toHaveBeenCalledWith(
+      { onAccept: cookieOnAllow, onReject: cookieOnReject, type: 'cookie' },
+      {},
+    );
+  });
```

Comment:
Think it would be good to have some tests that ensure that when the component is called with both set to false that `logic.not.toHaveBeenCalled` should pass

Reply:
Done :) We now have a test for both set, 1 set, other set, and none set :)